src/pydrex/visualisation.py

```diff
@@ -131,7 +134,104 @@ def polefigures(
     fig.savefig(_io.resolve_path(savefile))
 
 
-def alignment(ax, strains, angles, markers, labels, err=None, θ_max=90, θ_fse=None):
+def pathline_box2d(
+    ax,
+    get_velocity,
+    ref_axes,
+    colors,
+    positions,
+    marker,
+    min_coords,
+    max_coords,
+    resolution,
+    scale=1,
+    cmap=cmc.batlow,
+):
+    """Plot pathlines and velocity arrows for a 2D box domain.
+
+    If `ax` is None, a new figure and axes are created with `figure_unless`.
+
+    Args:
+    - `get_velocity` (callable) — object with call signature f(x) that returns
+      the 3D velocity vector at a given 3D position vector
+    - `ref_axes` (two letters from {"X", "Y", "Z"}) — labels for the horizontal and
+      vertical axes (these also define the projection for the 3D velocity/position)
+    - `colors` (array) — monotonic values along a representative pathline in the flow
+    - `positions` (Nx3 array) — 3D position vectors along the same pathline
+    - `min_coords` (array) — 2D coordinates of the lower left corner of the domain
+    - `max_coords` (array) — 2D coordinates of the upper right corner of the domain
+    - `resolution` (array) — 2D resolution of the velocity arrow grid (i.e. number of
+      grid points in the horizontal and vertical directions)
+    - `scale` (float, optional) — scale factor for the velocity arrows
+    - `cmap` (Matplotlib color map, optional) — color map for `colors`
+
+    Returns the figure handle, the axes handle, the quiver collection (velocities) and
+    the scatter collection (pathline).
+
+    """
+    fig, ax = figure_unless(ax)
+    ax.set_xlabel(ref_axes[0])
+    ax.set_ylabel(ref_axes[1])
+
+    x_min, y_min = min_coords
+    x_max, y_max = max_coords
+    ax.set_xlim((x_min, x_max))
+    ax.set_ylim((y_min, y_max))
+
+    x_res, y_res = resolution
+    if x_res == y_res:
+        ax.set_aspect("equal")
```

Comment:
Why did you make that choice? You could instead have a threshold on their ratio to include further scenarios. Equal aspect representation is helpful. :)